core/src/main/java/me/lokka30/treasury/plugin/core/utils/UpdateChecker.java

```diff
@@ -43,21 +46,75 @@ public static void checkForUpdates() {
         if (!plugin.configAdapter().getSettings().checkForUpdates()) {
             return;
         }
-        plugin.scheduler().runAsync(() -> {
-            try {
-                URL url = new URL(REQUEST_URI);
-                try (Reader in = new InputStreamReader(url.openStream())) {
-                    JsonObject object = Utils.GSON.fromJson(in, JsonObject.class);
-                    String latestVersion = object.get("current_version").getAsString();
-                    handleVersionCheck(latestVersion);
+        // since spigot is going to get only releases, we shall check whether this version is a
+        // release first and if it is a release, check spigot, otherwise check github commits
+        PluginVersion current = plugin.getVersion();
+        if (current.isReleaseVersion()) {
+            plugin.scheduler().runAsync(() -> {
+                try {
+                    URL url = new URL(SPIGOT_REQUEST_URI);
+                    try (Reader in = new InputStreamReader(url.openStream())) {
+                        JsonObject object = Utils.GSON.fromJson(in, JsonObject.class);
+                        String latestVersion = object.get("current_version").getAsString();
+                        handleSpigotVersionCheck(latestVersion);
+                    }
+                } catch (IOException e) {
+                    plugin.logger().error("IO whilst trying to request Spigot", e);
                 }
-            } catch (IOException e) {
-                plugin.logger().error("IO whilst trying to request Spigot", e);
-            }
-        });
+            });
+        } else if (current.isDevelopmentVersion()) {
+            plugin.scheduler().runAsync(() -> {
+                try {
+                    URL commits = new URL(GITHUB_COMMITS_URI);
+                    try (Reader commitsIn = new InputStreamReader(commits.openStream())) {
+                        JsonObject first = Utils.GSON
+                                .fromJson(commitsIn, JsonArray.class)
+                                .get(0)
+                                .getAsJsonObject();
+                        // we can't really compare SHA's as we have an abbreviated SHA rather
+                        // than the full commit SHA, that's why we're comparing dates
+                        OffsetDateTime latestTime = OffsetDateTime.parse(first
+                                .get("commit")
+                                .getAsJsonObject()
+                                .get("author")
+                                .getAsJsonObject()
+                                .get("date")
+                                .getAsString());
+
+                        // now get our commit date
+                        OffsetDateTime madeIn = OffsetDateTime.parse(UpdateChecker.class
+                                .getPackage()
+                                .getImplementationVersion());
```

Comment:
How does this get our commit date?

Reply:
I've configured the maven jar plugin to create a manifest file containing an entry "Implementation-Version" which is populated via maven with the commit date.

Reply:
I see